omod/src/main/java/org/openmrs/module/adminui/page/controller/account/ChangeDefaultsPageController.java

```diff
@@ -0,0 +1,146 @@
+import org.openmrs.module.adminui.AdminUiConstants;
+import org.openmrs.ui.framework.annotation.BindParams;
+import org.openmrs.ui.framework.annotation.MethodParam;
+import org.openmrs.ui.framework.annotation.SpringBean;
+import org.openmrs.ui.framework.page.PageModel;
+import org.openmrs.util.OpenmrsConstants;
+import org.springframework.context.MessageSource;
+import org.springframework.validation.BindingResult;
+import org.springframework.validation.ObjectError;
+
+import javax.servlet.http.HttpServletRequest;
+import java.util.List;
+import java.util.Map;
+
+public class ChangeDefaultsPageController {
+
+	public String get(PageModel pageModel) {
+		if (Context.isAuthenticated()) {
+			User user = Context.getAuthenticatedUser();
+			Map<String, String> props = user.getUserProperties();
+			UserDefaults userDefaults = new UserDefaults();
+			userDefaults.setDefaultLocale(props.get(OpenmrsConstants.USER_PROPERTY_DEFAULT_LOCALE));
+			userDefaults.setProficientLocales(props.get(OpenmrsConstants.USER_PROPERTY_PROFICIENT_LOCALES));
+			pageModel.addAttribute("userDefaults", userDefaults);
+			pageModel.addAttribute("locales", Context.getAdministrationService().getPresentationLocales());
+		}
+		return "account/changeDefaults";
+	}
+
+	public String post(@MethodParam("createUserDefaults") @BindParams UserDefaults defaults,
+	                   BindingResult errors,
+	                   @SpringBean("userService") UserService userService,
+	                   @SpringBean("messageSourceService") MessageSourceService messageSourceService,
+	                   @SpringBean("messageSource") MessageSource messageSource,
+	                   HttpServletRequest request,
+	                   PageModel model) {
+		if (errors.hasErrors()) {
+			sendErrorMessage(errors, messageSource, request);
+			model.addAttribute("errors", errors);
+			return "account/changePassword";
+		}
+		return saveDefaults(defaults, userService, messageSourceService, request);
+	}
+
+	private void sendErrorMessage(BindingResult errors, MessageSource messageSource, HttpServletRequest request) {
+		List<ObjectError> allErrors = errors.getAllErrors();
+		String message = getMessageErrors(messageSource, allErrors);
+		request.getSession().setAttribute(AdminUiConstants.SESSION_ATTRIBUTE_ERROR_MESSAGE, message);
+	}
+
+	private String getMessageErrors(MessageSource messageSource, List<ObjectError> allErrors) {
+		String message = "";
+		for (ObjectError error : allErrors) {
+			Object[] arguments = error.getArguments();
+			String errorMessage = messageSource.getMessage(error.getCode(), arguments, Context.getLocale());
+			message = message.concat(replaceArguments(errorMessage, arguments).concat("<br>"));
+		}
+		return message;
+	}
+
+	private String replaceArguments(String message, Object[] arguments) {
+		if (arguments != null) {
+			for (int i = 0; i < arguments.length; i++) {
+				String argument = (String) arguments[i];
+				message = message.replaceAll("\\{" + i + "\\}", argument);
```

Comment:
You should never have to replace place holders manually, this is automatically done by frameworks